pkg/api/store/cluster/cluster_store.go

```diff
@@ -207,15 +208,16 @@ func (r *Store) Create(apiContext *types.APIContext, schema *types.Schema, data
 	return r.Store.Create(apiContext, schema, data)
 }
 
-func loadDataFromTemplate(clusterTemplateRevision *v3.ClusterTemplateRevision, data map[string]interface{}) (map[string]interface{}, error) {
+func loadDataFromTemplate(clusterTemplateRevision *v3.ClusterTemplateRevision, clusterTemplate *v3.ClusterTemplate, data map[string]interface{}) (map[string]interface{}, error) {
 	dataFromTemplate, err := convert.EncodeToMap(clusterTemplateRevision.Spec.ClusterConfig)
 	if err != nil {
 		return nil, err
 	}
 	dataFromTemplate["name"] = convert.ToString(data["name"])
 	dataFromTemplate["description"] = convert.ToString(data[managementv3.ClusterSpecFieldDisplayName])
-	dataFromTemplate["clusterTemplateId"] = convert.ToString(data[managementv3.ClusterSpecFieldClusterTemplateID])
-	dataFromTemplate["clusterTemplateRevisionId"] = convert.ToString(data[managementv3.ClusterSpecFieldClusterTemplateRevisionID])
+	dataFromTemplate[managementv3.ClusterSpecFieldClusterTemplateID] = ref.Ref(clusterTemplate)
+	dataFromTemplate[managementv3.ClusterSpecFieldClusterTemplateRevisionID] = convert.ToString(data[managementv3.ClusterSpecFieldClusterTemplateRevisionID])
```

Comment:
Do we need to check if these fields are present before calling convert.ToString?

Reply:
@mrajashree I think convert already takes care of checking for nil/empty

Reply:
no, this method is called only after checking that the template exists in the data. cluster_store: hasTemplate() and cluster_store::validateTemplateInput validates the data[] input.

Reply:
yeah and convert return "" if passed nil to it.